addons/mail/static/src/components/discuss_sidebar/discuss_sidebar.js

```diff
@@ -116,7 +116,12 @@ class DiscussSidebar extends Component {
                 thread.isPinned &&
                 thread.model === 'mail.channel'
             )
-            .sort((c1, c2) => c1.displayName < c2.displayName ? -1 : 1);
+            .sort((c1, c2) => {
+                if (c1.displayName && c2.displayName && c1.displayName !== c2.displayName) {
+                    return c1.displayName.toLowerCase() < c2.displayName.toLowerCase() ? -1 : 1;
+                }
```

Comment:
This sort method is still not complete enough for me.
If c1 has displayName and c2 does not, I would except c1 to be first.
If c2 has displayName and c1 does not, I would except c2 to be first.
And only after that, if both have a displayName, we can compare them.

Reply:
@seb-odoo . How can it be possible that any channel does not have the displayName as displayName is the compute field based on the name.
I think all the channels having the channel_type = "channel" has a name field mandatory and so displayName is also.
[Refer This Method](https://github.com/odoo/odoo/blob/14.0/addons/mail/static/src/models/thread/thread.js#L1059)

Reply:
This is a very important point to understand, the JS works with incomplete information.
A field that is required in python is not guaranteed to have a value known in JS. It might be known is all current business flows but it is not guaranteed to be. As a dummy example, if you just do a `search([])` RPC it will return you the ids of matching channels, but not their names.

If you don't want to handle missing name in the sort method, another goods alternative would be to pre-filter on channels that have a name only in the filter/all that is done just above.

Note: in later version we introduced required field in JS too so that any missing info that we find important could raise an error in the first place. But I'm not even sure if channel name is one of them.

Reply:
This is a very important point to understand, the JS works with incomplete information.
A field that is required in python is not guaranteed to have a value known in JS. It might be known is all current business flows but it is not guaranteed to be. As a dummy example, if you just do a `search([])` RPC it will return you the ids of matching channels, but not their names.

If you don't want to handle missing name in the sort method, another goods alternative would be to pre-filter on channels that have a name only in the filter/all that is done just above.

Note: in later version we introduced required field in JS too so that any missing info that we find important could raise an error in the first place. But I'm not even sure if channel name is one of them.

Reply:
@seb-odoo . Thanks for the explanation. I have made the changes accordingly. Just review it.